### PR TITLE
Use Webpack bundle tracker v1

### PR DESCRIPTION
**Pull request by gilmrjc**
This PR adds support for webpack-bundle-tracker v1 and the new stats format.

### CODE REVIEW

File: webpack_loader/loader.py [hunk 1]
```diff
@@ -37,21 +37,22 @@ def get_assets(self):
...
             if not ignore:
-                chunk['url'] = self.get_chunk_url(chunk)
-                yield chunk
+                url = self.get_chunk_url(chunk, files)
+                yield { 'name': chunk, 'url': url }
 
-    def get_chunk_url(self, chunk):
-        public_path = chunk.get('publicPath')
+    def get_chunk_url(self, chunk, files):
+        chunk_file = files[chunk]
```
Comment (owais) [inline webpack_loader/loader.py:49]: If all this function is doing with files is to look up for a file using a name, I suggest it receive the file directly and the caller does the lookup.

File: webpack_loader/loader.py [hunk 2]
```diff
@@ -64,7 +65,7 @@ def get_bundle(self, bundle_name):
             timeout = self.config['TIMEOUT'] or 0
             timed_out = False
             start = time.time()
-            while assets['status'] == 'compiling' and not timed_out:
+            while (assets['status'] == 'compiling' or assets['status'] == 'compile') and not timed_out:
```
Comment (owais) [inline webpack_loader/loader.py:68]: suggestion: `assets['status'] in ['compile', 'compiling']`

File: webpack_loader/loader.py (hunk 1 above)
Comment (gilmrjc) [reply webpack_loader/loader.py]: Done, I kept the function signature as it was before.

File: tests/app/tests/test_webpack.py [hunk 3]
```diff
@@ -219,7 +217,7 @@ def test_request_blocking(self):
         # FIXME: This will work 99% time but there is no garauntee with the
         # 4 second thing. Need a better way to detect if request was blocked on
         # not.
-        wait_for = 3
+        wait_for = 4
```
Comment (michelts) [inline tests/app/tests/test_webpack.py:220]: Cool! I suppose this change is to match the comment. Can I suggest to fix the garauntee typo (that should be guarantee)?

File: tests/webpack.config.split.js [hunk 4]
```diff
@@ -15,26 +14,42 @@ module.exports = {
...
+        test: /\.jsx?$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'babel-loader',
+          options: {
+            presets: ['@babel/preset-react']
+          }
+        }
+      },
+      { test: /\.css$/, use: [MiniCssExtractPlugin.loader, 'css-loader'], }
```
Comment (michelts) [inline tests/webpack.config.split.js:34]:
```suggestion
      { test: /\.css$/, use: [MiniCssExtractPlugin.loader, 'css-loader'] }
```

File: webpack_loader/loader.py [hunk 5]
```diff
@@ -64,7 +67,7 @@ def get_bundle(self, bundle_name):
             timeout = self.config['TIMEOUT'] or 0
             timed_out = False
             start = time.time()
-            while assets['status'] == 'compiling' and not timed_out:
+            while assets['status'] in ['compile', 'compiling'] and not timed_out:
```
Comment (michelts) [inline webpack_loader/loader.py:70]: I'm not an expert in this area, I will suppose you know what you are doing when adding the *compile* status :)

Anyway, I think you should test this status just like you test the *compiling* one. I found occurrences of the compiling status in the following places:

The test_request_blocking func:

https://github.com/gilmrjc/django-webpack-loader/blob/webpack-bundle-tracker-1/tests/app/tests/test_webpack.py#L195

The test_timeouts func:

https://github.com/gilmrjc/django-webpack-loader/blob/webpack-bundle-tracker-1/tests/app/tests/test_webpack.py#L224

File: tests/package.json [hunk 6]
```diff
@@ -6,17 +6,16 @@
...
-    "css-loader": "^0.14.1",
-    "extract-text-webpack-plugin": "^0.8.0",
-    "node-libs-browser": "^0.5.0",
-    "react": "^0.13.3",
-    "style-loader": "^0.12.3",
-    "webpack": "^1.9.8",
-    "webpack-bundle-tracker": "0.0.8",
-    "webpack-dev-server": "^1.9.0",
-    "webpack-split-by-path": "0.0.1"
+    "@babel/core": "^7.9.6",
```
Comment (michelts) [inline tests/package.json:9]: The PR is there for a while, some packages were updated in the npm registry. Should we upgrade them?

File: tests/app/tests/test_webpack.py (hunk 3 above)
Comment (gilmrjc) [reply tests/app/tests/test_webpack.py:220]: Sure, I'll do it.

File: tests/package.json (hunk 6 above)
Comment (gilmrjc) [reply tests/package.json:9]: The purpose of this PR is to test the new version of `webpack-bundle-tracker` which is in alpha state right now. To me it makes sense trying this integration and set the `django-webpack-loader` in alpha state. Once this is know to be working, bump the dependencies during the official new release. Is that ok?

File: webpack_loader/loader.py (hunk 5 above)
Comment (gilmrjc) [reply webpack_loader/loader.py:70]: Actually the `compiling` state was removed in favor of `compile`.

You can see it was removed here:
https://github.com/owais/webpack-bundle-tracker/commit/d1f8e39c0c5bde252c1025174607ebf048ca9f94#diff-6d186b954a58d5bb740f73d84fe39073L60

And changed here:
https://github.com/owais/webpack-bundle-tracker/commit/d1f8e39c0c5bde252c1025174607ebf048ca9f94#diff-6d186b954a58d5bb740f73d84fe39073R102

I will remove the `compiling` reference. Thanks

File: tests/package.json (hunk 6 above)
Comment (michelts) [reply tests/package.json:9]: You are right, it is not a concern for this PR :)

Sorry about that.

**Comment by pydolan** [comment]
What else is left on this PR? I can help if there's any actionable items left

**Comment by agustinf88** [comment]
Any update on this PR? world be great to have this version

**Comment by devopsjosh** [comment]
> Any update on this PR? world be great to have this version

I think they may be waiting on more feedback for features since this is a breaking change. However, yeah this would be nice to have ASAP!

https://github.com/owais/django-webpack-loader/issues/215

**Comment by DolanMaize** [comment]
Good point about this being a breaking change.  How about adding a setting to have this disabled by default (thus not breaking), and eventually (several months from now?) the default can change

**Comment by gilmrjc** [comment]
As far as I understand this a *expected* breaking change. This is ready to go, but @owais has no merged it. Let's have patience and wait for him.

**Review by andersk — APPROVED** [review]
This works for me in Zulip (zulip/zulip#16653).

**Comment by zodman** [comment]
it works for me thanks!

**Review by zodman — APPROVED** [review]

**Review by paltman — APPROVED** [review]

**Comment by qguv** [comment]
Hi @owais, it looks like this is ready to go. Any plans to merge this soon? I think it blocks the upgrade to webpack v5.

**Comment by owais** [comment]
Thanks for all the contributions and patience everyone. Unfortunately I cannot merge the PR until I can test, verify and understand the changes completely and I really don't have any free time to do that. I know this project has suffered quite a bit because of me moving on to other things and I'd like to pass it on to people that actually use this piece of software every day and are willing to maintain it. I've updated the readme with a message to reflect tihs. 

If your organization are heavy users of django-webpack-loader and webpack-bundle-tracker, I'd love to invite you to help maintain the project and transfer both the repositories, pypi and npm packages to you. If it sounds like something your organization would like to do, please get in touch with me over email. Of course, individuals are welcome as well especially with existing track record of maintaining OSS software.

**Review by fjsj — APPROVED** [review]

**Pull request merged**